saleor/checkout/utils.py

```diff
@@ -238,12 +248,15 @@ def add_variants_to_checkout(
     return checkout
 
 
-def _append_line_to_update(
-    to_update, to_delete, variant, line_data, replace, variant_ids_in_lines
-):
-    if variant.pk not in variant_ids_in_lines:
+def _get_line_if_exist(line_data, lines_by_ids):
+    if line_data.line_id and line_data.line_id in lines_by_ids:
```

Comment:
Is it possible that `line_id` will be None?
Also, can't we replace this function by `lines_by_ids.get(line_data.line_id)`? 
or `lines_by_ids.get(line_data.line_id) if line_data.line_id else None`

Reply:
yes, it possible, on `linesAdd` line id can be `None`, but you are right I will change this function to one line